doc/validation.md

```diff
@@ -0,0 +1,126 @@
+Validation rules in `easy-deposit-ui`
+=====================================
+
+This document describes the various validation rules that are applied to the forms in `easy-deposit-ui`.
+The errors caught in the validation process are accumulated and presented to the user on screen.
+
+
+Login form
+----------
+
+* **username**
+  * mandatory field
+    * _error text:_ "no username was provided"
+* **password**
+  * mandatory field
+    * _error text:_ "no password was provided"
+
+
+Deposit form
+------------
+
+### Basic information form
+
+* **doi**
+  * mandatory field
+    * _error text:_ "no doi was provided"
+* **languageOfDescription**
+  * mandatory field
+    * _error text:_ "no language of description was provided"
+* **titles**
+  * mandatory field array
+    * _error text:_ "no titles were provided"
+* **description**
+  * mandatory field
+    * _error text:_ "no description was provided"
+* **contributors**
+  * mandatory field array
+    * _error text:_ "no contributors were provided"
+  * one creator minimum
+    * _error text:_ "at least one creator is required"
+  * every nonempty contributor must have either an organization or {initial, surname}
```

Comment:
When do you return which message? They don't reflect the rule. Same for RightsHolders.

See also 
https://github.com/DANS-KNAW/easy-deposit-api/blob/abbcc43cadb5a2e964cafb6d27e60fe1670541ec/src/test/scala/nl.knaw.dans.easy.deposit/servlets/ValidationSpec.scala#L202-L222

Reply:
Hopefully f5b58f3 made this somewhat more clear. However, I don't get what you mean with the `ValidationSpec` reference.

Reply:
With ValidationSpec I wanted to show how I validate and report certain input.